src/pyapi.jl

```diff
@@ -5,73 +5,101 @@
 using PyCall
 using Pandas
 
-# Took this from the example file. Unclear what this should be, or should be an argument
-const DEFAULT_CONFIG = "../repos/data_pipeline_api/examples/test_data_2/config.yaml"
-
 function pycallinit()
     py"""
     import pandas as pd
+    import scipy.stats
     from pathlib import Path
     from data_pipeline_api.simple_network_sim_api import SimpleNetworkSimAPI
     from data_pipeline_api.standard_api import StandardAPI
     print(SimpleNetworkSimAPI)
 
-    def read_estimate(data, component):
+    def read_estimate(api, data, component):
         print("[PYTHON]: Reading estimate")
-        with StandardAPI(
-             $DEFAULT_CONFIG,
-        ) as api:
-            d = api.read_estimate(data, component)
-
-        return d
+        return api.read_estimate(data, component)
 
-    def read_distribution(data, component):
+    def read_distribution(api, data, component):
         print("[PYTHON]: Reading distribution")
-        with SimpleNetworkSimAPI(
-             $DEFAULT_CONFIG,
-        ) as api:
-            d = api.read_distribution(data)
+        return api.read_distribution(data, component)
 
-        return d
+    def read_sample(api, data, component):
+        print("[PYTHON]: Reading sample")
+        return api.read_sample(data, component)
 
-    def read_array(data, component):
+    def read_array(api, data, component):
         print("[PYTHON]: Reading array")
-        with SimpleNetworkSimAPI(
-             $DEFAULT_CONFIG,
-        ) as api:
-            d = api.read_array(data)
+        return api.read_array(data, component)
 
-        return d
-
-    def read_table(data, component):
+    def read_table(api, data, component):
         print("[PYTHON]: Reading table")
-        with SimpleNetworkSimAPI(
-             $DEFAULT_CONFIG,
-        ) as api:
-            d = api.read_table(data)
-
-        return d
+        return api.read_table(data, component)
     """
+end
+
+abstract type FileAPI end
 
+struct StandardAPI <: FileAPI
+    pyapi::PyObject
 end
 
+struct SimpleNetworkSimAPI <: FileAPI
+    pyapi::PyObject
+end
 
-function read_estimate(data_product, component)
-    d = py"read_estimate($data_product, $component)"
-    return d
+function StandardAPI(f::Function, config_filename)
```

Comment:
This is where the magic happens so a few comments here would be helpful.

Reply:
Yep, I added some docstrings and comments.